src/basic/math/FCOMP.s

```diff
@@ -0,0 +1,92 @@
+;; #LAYOUT# STD *       #TAKE
+;; #LAYOUT# X16 BASIC_0 #TAKE-OFFSET 2000
+;; #LAYOUT# *   BASIC_0 #TAKE
+;; #LAYOUT# *   *       #IGNORE
+
+;
+; Math package - compare FAC1 with RAM location
+;
+; Input:
+; - .A - address low byte
+; - .Y - address high byte
+;
+; Output:
+; - .A - $00 for equal, $01 for FAC1 > RAM, $FF for FAC1 < RAM
+; - flags N/Z, depending on whether FAC1 is 0, positive, or negative
+;
+; Notes:
+; - uses vector at $24, leaves FAC1 and FAC2 unchanged (Codebase64)
+;
+; See also:
+; - [CM64] Computes Mapping the Commodore 64 - page 115
+; - https://www.c64-wiki.com/wiki/Floating_point_arithmetic
+; - https://codebase64.org/doku.php?id=base:kernal_floating_point_mathematics
+;
+
+
+FCOMP: 
+    sta INDEX+2
+    sty INDEX+3
+
+    jsr sgn_FAC1_A          ; Set N/Z flag for FAC1 and store on stack
+    php
+
+    ldy #$01
+    lda #$7F                ; Expand sign of RAM float into A as $FF or $00
+    cmp (INDEX+2),Y
+    adc #$80
+    cmp FAC1_sign
+    bne greater_than         ; FAC < or > RAM, will be swapped to correct
+
+    dey
+    lda (INDEX+2),Y         ; Exponent of RAM float into A
+    cmp FAC1_exponent
+    bcc greater_than        ; FAC1 > RAM
+    bne less_than
+
+    cmp #$00
+    beq equal               ; FAC1 = RAM = 0
+
+    iny
+    lda (INDEX+2),Y
+    ora #$80                ; Set highest bit of mantissa (0.5)
+    cmp FAC1_mantissa
+    bcc greater_than
+    bne less_than
+    iny
+    lda (INDEX+2),Y
+    cmp FAC1_mantissa+1
+    bcc greater_than
+    bne less_than
+    iny
+    lda (INDEX+2),Y
+    cmp FAC1_mantissa+2
+    bcc greater_than
+    bne less_than
+    iny
+    lda (INDEX+2),Y
+    cmp FAC1_mantissa+3
+    bcc greater_than
+    bne less_than
+
+equal:
```

Comment:
Rename this to `FCOMP_equal` (similarly with other symbols); unfortunately, these symbols are exported and visible outside (so you can't have another label called `equal` anywhere else) - and we don't want naming conflicts.